packages/core/src/config.ts

```diff
@@ -171,13 +171,13 @@ export async function resolveConfig<Theme extends object = object>(
       if (isStaticRule(rule)) {
         const prefixes = toArray(rule[2]?.prefix || '')
         prefixes.forEach((prefix) => {
-          rulesStaticMap[prefix + rule[0]] = [i, rule[1], rule[2], rule]
+          rulesStaticMap[prefix + rule[0]] = [i, rule[0], rule[1], rule[2], rule]
         })
         // delete static rules so we can't skip them in matching
         // but keep the order
         return undefined
       }
-      return [i, ...rule]
+      return [i, rule[0], rule[1], rule[2], rule]
```

Comment:
I wonder if we can just return `[i, rule]`, or even the `i` is not needed?

Reply:
I made a PR to this branch with https://github.com/unocss/unocss/pull/4304

Seems to work fine?